rinja_parser/src/node.rs

```diff
@@ -1202,3 +1202,99 @@ impl<'a> Comment<'a> {
 /// Second field is "minus/plus sign was used on the right part of the item".
 #[derive(Clone, Copy, Debug, PartialEq)]
 pub struct Ws(pub Option<Whitespace>, pub Option<Whitespace>);
+
+fn is_rust_keyword(ident: &str) -> bool {
+    const MAX_KW_LEN: usize = 8;
+
+    const KW0: &[[u8; MAX_KW_LEN]] = &[];
+    const KW1: &[[u8; MAX_KW_LEN]] = &[];
+    const KW2: &[[u8; MAX_KW_LEN]] = &[
+        *b"as______",
+        *b"do______",
+        *b"fn______",
+        *b"if______",
+        *b"in______",
+    ];
+    const KW3: &[[u8; MAX_KW_LEN]] = &[
+        *b"box_____",
+        *b"dyn_____",
+        *b"for_____",
+        *b"let_____",
+        *b"mod_____",
+        *b"mut_____",
+        *b"pub_____",
+        *b"ref_____",
+        *b"try_____",
+        *b"use_____",
+    ];
+    const KW4: &[[u8; MAX_KW_LEN]] = &[
+        *b"else____",
+        *b"enum____",
+        *b"impl____",
+        *b"loop____",
+        *b"move____",
+        *b"priv____",
+        *b"self____",
+        *b"Self____",
+        *b"true____",
+        *b"type____",
+    ];
+    const KW5: &[[u8; MAX_KW_LEN]] = &[
+        *b"async___",
+        *b"await___",
+        *b"break___",
+        *b"const___",
+        *b"crate___",
+        *b"false___",
+        *b"final___",
+        *b"macro___",
+        *b"match___",
+        *b"super___",
+        *b"trait___",
+        *b"union___",
+        *b"where___",
+        *b"while___",
+        *b"yield___",
+    ];
+    const KW6: &[[u8; MAX_KW_LEN]] = &[
+        *b"become__",
+        *b"extern__",
+        *b"return__",
+        *b"static__",
+        *b"struct__",
+        *b"typeof__",
+        *b"unsafe__",
+    ];
+    const KW7: &[[u8; MAX_KW_LEN]] = &[*b"unsized_", *b"virtual_"];
+    const KW8: &[[u8; MAX_KW_LEN]] = &[*b"abstract", *b"continue", *b"override"];
+
+    const KWS: &[&[[u8; MAX_KW_LEN]]] = &[KW0, KW1, KW2, KW3, KW4, KW5, KW6, KW7, KW8];
+
+    // Ensure that all strings are ASCII, because we use `from_utf8_unchecked()` further down.
+    const _: () = {
```

Comment:
Instead of making it a const check at every compilation, wouldn't it better to have it in a `#[test]`? You can add a test inside this function directly to use the exact same code.

Reply:
Done.